File: ProspectManagerWebApi/DTO/Request/ProspectSearchRequestDTO.cs
using ProspectManagerWebApi.Models;

namespace ProspectManagerWebApi.DTO.Request
{
    public class ProspectSearchRequestDTO
    {
        public string[] Noms { get; set; }
        public Statut[] Statuts { get; set; }
        public string[] Departements { get; set; }
        public string[] SecteursActivite { get; set; }
        public Produit[] Produits { get; set; }
        public TypeOrganisme[] TypesOrganisme { get; set; }
    }
}

File: ProspectManagerClient/src/app/modules/gestion-prospect/dto/prospect-search-request-dto.ts
import { Produit } from "../models/produit";
import { SecteurGeographique } from "../models/secteur-geographique";
import { Statut } from "../models/statut";
import { TypeOrganisme } from "../models/type-organisme";

export class ProspectSearchRequestDto {
    noms: string[] = [];
    statuts: Statut[] = [];
    secteursGeographiques: SecteurGeographique[] = [];
    secteursActivite: string[] = [];
    produits: Produit[] = [];
    typesOrganisme: TypeOrganisme[] = [];
    inclureInactif: boolean = false;
}